Makefile: fix vendor modvendor

otan opened the pull request:
Since modvendor requires installation during vendor rebuilding, but a
dependency may have changed, `go mod` complains about inconsistent
versioning. To fix this, use -mod=mod when installing modvendor.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50802)
<!-- Reviewable:end -->

otan committed:
Makefile: fix vendor modvendor

Since modvendor requires installation during vendor rebuilding, but a
dependency may have changed, `go mod` complains about inconsistent
versioning. To fix this, use -mod=mod when installing modvendor.

Release note: None

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50802#-:-MB4HXikAiE77JTWBbbQ:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jordanlewis and @otan)

___
*[Makefile, line 319 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50802#-MB4HPXU8__vYOAGnJik:-MB4HPXV-NPS5iSYgxjr:bmtt2xu) ([raw file](https://github.com/cockroachdb/cockroach/blob/1895f7080f42eddfb04d69f3c4eb3be5ca234c9f/Makefile#L319)):*
> ```Makefile
> 	# Use -mod=mod, as -mod=vendor will try install from the vendor directory
> 	# which may be mismatching upon rebuild.
> 	$(GO_INSTALL) -v -mod=mod github.com/goware/modvendor
> ```

Since we're not using the vendored version of `modvendor`, do we need `cmd/import-tools/main.go` to import `github.com/goware/modvendor`?


<!-- Sent from Reviewable.io -->

otan commented:
___
*[Makefile, line 319 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50802#-MB4HPXU8__vYOAGnJik:-MB4_JC7-CnXNs18NbBw:b-y1e1f4) ([raw file](https://github.com/cockroachdb/cockroach/blob/1895f7080f42eddfb04d69f3c4eb3be5ca234c9f/Makefile#L319)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Since we're not using the vendored version of `modvendor`, do we need `cmd/import-tools/main.go` to import `github.com/goware/modvendor`?
</blockquote></details>

-mod=mod uses the version in `go.mod` (it just doesn't try to read the vendor directory!)


<!-- Sent from Reviewable.io -->

otan commented:
bors r=petermattis
